util/cosmos.js

```diff
@@ -1,43 +1,29 @@
-import axios from 'axios';
-import { timeout } from '../common/util/misc';
+import { StargateClient } from '@cosmjs/stargate';
 
-const api = axios.create({
-  baseURL: '/api/proxy/cosmos/txs',
-  headers: {
-    'Content-Type': 'application/json',
-  },
-});
+import { COSMOS_ENDPOINT } from '../server/config/config';
+import { timeout } from '../common/util/misc';
 
-export async function waitForTxToBeMined(txHash) {
+export default async function waitForTxToBeMined(txHash) {
   let tx;
   let notFoundOnce = false;
+  const client = await StargateClient.connect(COSMOS_ENDPOINT);
   while (!tx) {
     /* eslint-disable no-await-in-loop */
     await timeout(1000);
     try {
-      const { data } = await api.get(`/${txHash}`);
-      if (data && data.height) {
+      const data = await client.getTx(txHash);
+      if (data) {
```

Comment:
Note: `data` will either be `null` or type `IndexedTx`, ref [here](https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/stargateclient.ts#L286).